JavaScript/JavaScriptSDK/snippet.js

```diff
@@ -78,4 +78,8 @@
 
 // global instance must be set in this order to mitigate issues in ie8 and lower
 window.appInsights = appInsights;
-appInsights.trackPageView();
+
+// if somebody calls the snippet twice, don't report page view again
+if (appInsights.queue && appInsights.queue.length === 0) {
```

Comment:
Just curious, may we have some race condition in case of a single snippet, so that queue.length is 0 at this time and not even the first TrackPageView() is called, e.g. for the first Page View?